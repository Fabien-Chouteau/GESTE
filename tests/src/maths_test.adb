with Ada.Text_IO;       use Ada.Text_IO;

with GESTE.Maths_Types; use GESTE.Maths_Types;
with GESTE.Maths;       use GESTE.Maths;

procedure Maths_Test is

   Sqrt_Values : array (Natural range <>) of Value :=
     (0.0, 1.0, 1.00001, 2.0, 4.0, 25.0, 80.0, 81.0, 100.0);

   Cos_Values : array (Natural range <>) of Value :=
     (0.0, Pi, Pi / 2.0, -Pi / 2.0, Pi / 3.0, Pi * 10.0);

   Sin_Values : array (Natural range <>) of Value :=
     (0.0, Pi, Pi / 2.0, -Pi / 2.0, Pi / 6.0, Pi * 10.0);

   To_Rad_Values : array (Natural range <>) of Value :=
     (0.0, 180.0, 360.0);

   To_Degrees_Values : array (Natural range <>) of Value :=
     (0.0, Pi, Pi * 2.0, Pi * 10.0);
begin
   Put_Line ("Value'First'Img => " & Value'First'Img);
   Put_Line ("Value'Last'Img => " & Value'Last'Img);
   Put_Line ("Value'Small'Img => " & Value'Small'Img);


   for X of Sqrt_Values loop
      Put_Line ("Sqrt (" & X'Img & ") => " & Sqrt (X)'Img);
   end loop;

   for X of Cos_Values loop
      Put_Line ("Cos (" & X'Img & ") => " & Cos (X)'Img);
   end loop;

   for X of Sin_Values loop
      Put_Line ("Sin (" & X'Img & ") => " & Sin (X)'Img);
   end loop;

   for X of To_Rad_Values loop
      Put_Line ("To_Rad (" & X'Img & ") => " & To_Rad (X)'Img);
   end loop;

   for X of To_Degrees_Values loop
      Put_Line ("To_Degrees (" & X'Img & ") => " & To_Degrees (X)'Img);
   end loop;

   Put_Line ("Magnitude (Vect (0.0, 5.0)) =>" &
               Magnitude (Vect'(0.0, 5.0))'Img);

   Put_Line ("Magnitude (Vect (5.0, 0.0)) =>" &
               Magnitude (Vect'(5.0, 0.0))'Img);

   Put_Line ("Magnitude (Vect (5.0, 5.0)) =>" &
               Magnitude (Vect'(5.0, 5.0))'Img);


end Maths_Test;
