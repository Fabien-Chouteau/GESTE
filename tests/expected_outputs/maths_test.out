Value'First'Img => -2097152.0000
Value'Last'Img =>  2097151.9990
Value'Small'Img =>  9.76562500000000000E-04
Sqrt ( 0.0000) => -1.0000
Sqrt ( 1.0000) =>  1.0000
Sqrt ( 1.0000) =>  1.0000
Sqrt ( 2.0000) =>  1.4141
Sqrt ( 4.0000) =>  2.0000
Sqrt ( 25.0000) =>  5.0000
Sqrt ( 80.0000) =>  8.9434
Sqrt ( 81.0000) =>  9.0000
Sqrt ( 100.0000) =>  10.0000
Cos ( 0.0000) =>  1.0000
Cos ( 3.1406) => -1.0000
Cos ( 1.5703) =>  0.0000
Cos (-1.5703) =>  0.0000
Cos ( 1.0469) =>  0.4922
Cos ( 31.4063) =>  1.0000
Sin ( 0.0000) =>  0.0000
Sin ( 3.1406) =>  0.0000
Sin ( 1.5703) =>  1.0000
Sin (-1.5703) => -1.0000
Sin ( 0.5234) =>  0.4922
Sin ( 31.4063) =>  0.0000
To_Rad ( 0.0000) =>  0.0000
To_Rad ( 180.0000) =>  3.1406
To_Rad ( 360.0000) =>  6.2813
To_Degrees ( 0.0000) =>  0.0000
To_Degrees ( 3.1406) =>  180.0000
To_Degrees ( 6.2813) =>  360.0000
To_Degrees ( 31.4063) =>  1800.0000
Magnitude (Vect (0.0, 5.0)) => 5.0000
Magnitude (Vect (5.0, 0.0)) => 5.0000
Magnitude (Vect (5.0, 5.0)) => 7.0703
